src/soep_preparation/dataset_merging/task.py

```diff
@@ -33,216 +58,147 @@ def _dataset_level(dataset: pd.DataFrame) -> str:
-
-    Args:
-        datasets (dict): Dictionary containing the datasets to be merged.
-
-    Returns:
-        dict: Dictionary containing the datasets sorted by variation level.
-    """
-    datasets_by_level = {}
-    for name, dataset in datasets.items():
-        level = _dataset_level(dataset)
-        if level not in datasets_by_level:
-            datasets_by_level[level] = {name: dataset}
+    out = {
+        "individual-time-varying": {},
+        "individual-time-constant": {},
+        "household-time-varying": {},
+        "household-time-constant": {},
+    }
+    for name in names:
+        df = name_to_df_mapping[name]
+        df_level = _get_df_level(df)
+        out[df_level][name] = [
+            var for var in variables_not_merge_keys if var in df.columns
+        ]
+    return out
+
+
+def merge_data_by_level(
+    name_to_df_mapping: dict[str, pd.DataFrame],
+    name_to_variables_mapping: dict[str, list],
+    how: str = "outer",
+) -> pd.DataFrame:
+    out = None
+    for name, variables in name_to_variables_mapping.items():
+        columns = [
+            var for var in MERGE_KEYS if var in name_to_df_mapping[name].columns
+        ] + variables
+        if out is None:
+            out = name_to_df_mapping[name][columns]
         else:
-            datasets_by_level[level][name] = dataset
-    # Change order of datasets to make sure datasets with most reliable information on
-    # hh_id are merged first
-    datasets_by_level["individual-time-varying"] = _reorder_dict(
-        datasets_by_level["individual-time-varying"],
-        ["pequiv", "pgen", "pl"],
-    )
-    return datasets_by_level
-
+            shared_columns = _get_shared_columns(out, name_to_df_mapping[name])
+            right = name_to_df_mapping[name][columns]
+            out = out.merge(right, on=shared_columns, how=how)
+    return out
 
-def _add_origin_dummy(
-    datasets: dict[str, dict[str, pd.DataFrame]],
-) -> dict[str, pd.DataFrame]:
-    for dataset_group in datasets:
-        for dataset in datasets[dataset_group]:
-            datasets[dataset_group][dataset][f"from_{dataset}"] = pd.Series(
-                [1] * len(datasets[dataset_group][dataset]),
-                dtype="uint8[pyarrow]",
-            )
-    return datasets
 
-
-def combine_similar_information_from_different_columns(
-    data_merged: pd.DataFrame,
+def merge_data(
+    datasets: list[pd.DataFrame],
+    how: str = "outer",
 ) -> pd.DataFrame:
-    """Combining variables from different datasets with similar information.
-
-    Args:
-        data_merged (pd.DataFrame): Merged dataset.
+    out = None
+    for df in datasets:
+        if out is None:
+            out = df
+        elif df is not None:
+            shared_columns = _get_shared_columns(out, df)
+            out = out.merge(df, on=shared_columns, how=how)
+    return out
+
+
+def dataset_creation(
+    variables: list[str],
+    min_survey_year: int,
```

Comment:
```suggestion
    min_and_max_survey_years: tuple[int, int] | None = None,
    survey_years: list[int] | None = None,
```
Then add a check for sanity (max >= min etc.) and that exactly one of the two is not None.